src/IR/IR.hs

```diff
@@ -383,8 +411,97 @@ instance Pretty PrimOp where
   pretty PrimBitAnd = pretty "&"
   pretty PrimBitOr  = pretty "|"
   pretty PrimEq     = pretty "=="
-  pretty PrimNeq     = pretty "!="
+  pretty PrimNeq    = pretty "!="
   pretty PrimGt     = pretty ">"
   pretty PrimGe     = pretty ">="
   pretty PrimLt     = pretty "<"
   pretty PrimLe     = pretty "<="
+
+{- | Format Typeclass
+
+Runs on top of pretty typeclass to make Doc output more readable.
+Adds 
+- indentation and line breaks
+- parens where necessary
+Omits 
+- let _ =
+- type annotations
+Reverts
+- curried funcs of one arg back to multiple arg funcs
+-}
+
+class (Pretty a) => FMT a where
```

Comment:
`FMT` should be stylized as `Fmt`

But also, this looks like what `Pretty` should actually be? Is there a reason to introduce another typeclass?